analyses/mwi/dryspells_trigger/exploration/test_arc2_processing.md

```diff
@@ -0,0 +1,92 @@
+## ARC2 processing
+
+This notebook is meant to provide simple examples and a space for testing the ARC2 processing script as it's developed and until a more proper space for examples is built/designed. The idea behind the processing is that the ARC2 class is defined for a specific country, date range, and geographical bounds. Each time the class is instantiated, the processes do all the work behind the scenes to load in and update new raw data and process it, maintaining all within a master file. Let's first look at just the raw data.
+
+```python
+%load_ext autoreload
+%autoreload 2
+```
+
+```python
+import os
+from pathlib import Path
+import sys
+from datetime import date
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[2]}/"
+sys.path.append(path_mod)
+from src.indicators.drought.arc2_precipitation import DrySpells
+from src.utils_general.raster_manipulation import compute_raster_statistics
+
+
+poly_path = os.path.join(
+    os.getenv('AA_DATA_DIR'),
+    'public',
+    'raw',
+    'mwi',
+    'cod_ab',
+    'mwi_adm_nso_20181016_shp',
+    'mwi_admbnda_adm2_nso_20181016.shp'
+)
+```
+
+### Raw data downloading
+
+So, first, let's just get a class and download data for the 2 days specified. Since we are specifying the class `DrySpells` rather than just `ARC2`, it will automatically ensure sufficient data is downloaded to calculate the rolling sum (and potential dry spells) from the start to end of our monitoring period.
+
+```python
+arc2_test = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = "2021-09-02",
+    monitoring_end = "2021-09-03",
+    range_x = ("32E", "36E"),
+    range_y = ("20S", "5S")
+)
+
+arc2_test.download_data(master=True)
```

Comment:
If I change this to False and set the date range to one that is not yet present in the master file, I get the error `TypeError: unhashable type: 'list'`